### PR TITLE
Add forms pagination and see all link

### PR DESCRIPTION
**Pull request by donaldkibet**
### What does this PR do?

1. Adds forms-pagination
2. Ability to navigate away when `See all` link is clicked
3. Add patientChart pagination.

### To od
- [ ] fix broken test because the mocked [openmrs-esm-framework/mock](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/esm-framework/mock.tsx) interpolateString function returns `undefined` resulting in the test to fail

### Screenshoot
screencast 2021-03-19 16-20-22

### CODE REVIEW

File: src/ui-components/pagination/pagination.component.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  onPageNumberChange,
+  pageNumber,
+  pageUrl
+}) => {
+  const { t } = useTranslation();
+  const [, , patientUuid] = useCurrentPatient();
+  const chartBasePath = "${openmrsSpaBase}/patient/:patientUuid/chart/".replace(
+    ":patientUuid",
+    patientUuid
+  );
```
Comment (brandones) [inline src/ui-components/pagination/pagination.component.tsx]:
```suggestion
  const chartBasePath = `${openmrsSpaBase}/patient/${patientUuid}/chart/`;
```

File: src/ui-components/pagination/pagination.component.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  const generatePageSizes = () => {
+    const numberOfPages = Math.ceil(items?.length / pageSize);
+    return [...Array(numberOfPages).keys()].map(x => {
+      return ++x * pageSize;
```
Comment (brandones) [inline src/ui-components/pagination/pagination.component.tsx]:
```suggestion
      return (x + 1) * pageSize;
```

Makes my brain hurt less

File: src/ui-components/pagination/pagination.component.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    });
+  };
+
+  return (
+    <>
+      {items.length > 0 && (
```
Comment (brandones) [inline src/ui-components/pagination/pagination.component.tsx:45]: I might be wrong, but I think React will get mad if you try to render `false`. I think you want `items.length ? whatever : null`.

Comment (donaldkibet) [reply src/ui-components/pagination/pagination.component.tsx:45]: When using `Logical && Operator` if the condition is true, the element right after `&&` will appear in the output. If it is false, React will ignore and skip it.

**Review by brandones — APPROVED** [review]
LGTM, once failing checks are fixed

**Pull request merged**